### PR TITLE
[CLB-48] Add CRUD methods for clients endpoint.

### PR DESCRIPTION
**Pull request by SreckoStojic**

### CODE REVIEW

File: src/api/client/controllers.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    res.status(200).send(clients);
+  } catch (e) {
+    console.log(e);
+    res.sendStatus(500);
+  }
+};
+
+const addClient = async (req, res) => {
+  if (!req.body || !req.body.companyName) {
+    return res.status(400).send('Project name cannot be empty');
```
Comment (petarjs) [inline src/api/client/controllers.js]: Perhaps `Project name....` should be `Company name`?

File: src/api/client/controllers.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  }
+
+  const payload = {
+    companyName,
+    ...optionalFields,
+  };
+
+  try {
+    const createdClient = await clientService.addNewClient(payload);
+    const { ...retVal } = createdClient.dataValues;
```
Comment (petarjs) [inline src/api/client/controllers.js]: Is this the same as this :)

```suggestion
    const retVal = createdClient.dataValues;
```

File: src/api/client/controllers.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    return res.status(400).send('Missing ID param');
+  }
+
+  if (!req.body.client) {
+    return res.status(400).send('Client not found');
+  }
+
+  const client = req.body.client;
+
+  const { ...retVal } = client.dataValues;
```
Comment (petarjs) [inline src/api/client/controllers.js]:
```suggestion
  const retVal = client.dataValues;
```

File: src/api/client/params.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const clientService = require('../../services/client');
+
+const checkClient = async (req, res, next, id) => {
+  await clientService
+    .findOne({ id })
+    .then(function (client) {
```
Comment (petarjs) [inline src/api/client/params.js]: Use async await instead of then :)

**Review by petarjs — CHANGES_REQUESTED** [review]

File: src/api/client/controllers.js (hunk 1 above)
Comment (SreckoStojic) [reply src/api/client/controllers.js]: Yes, i will change it.

File: src/api/client/controllers.js (hunk 2 above)
Comment (SreckoStojic) [reply src/api/client/controllers.js]: I was thinking in way if i should hide something in response (like password in user), but after that i removed that part and forgot to change. I will fix it. Thanks.

File: src/api/client/params.js (hunk 4 above)
Comment (SreckoStojic) [reply src/api/client/params.js]: Sure, i will change it.

**Pull request merged**